[SPIKE] swapping busboy for multer

ckousik opened the pull request:
```
    new file: core/server/middleware/ghost-multer.js 
modified:   core/server/middleware/index.js
modified:   core/server/routes/api.js
modified:   package.json
```

Multer works fine when I run this on localhost

acburdine commented on core/server/middleware/index.js:
```diff
@@ -34,6 +35,7 @@ var bodyParser      = require('body-parser'),
 
 middleware = {
     busboy: busboy,
```
Wouldn't this line & the other related busboy code be removed?

acburdine commented on package.json:
```diff
@@ -52,6 +52,7 @@
     "lodash": "3.10.1",
     "moment": "2.11.1",
     "morgan": "1.6.1",
+    "multer": "^1.1.0",
```
This should be locked to a specific version: aka remove the `^`

acburdine commented:
Hey @ckousik! Thanks for contributing!

From a quick glance, this looks good! (Haven't tested it in practice though).

Couple comments (excluding the other ones I made):
- can you change the commit message to this: `deps: multer@1.1.0`
- the reason tests are failing is you have some linting errors: if you run `grunt lint` locally it will show you your errors
- if this truly removes busboy, all busboy-related code and references should be deleted
- it would be cool if you could try and write some sort of tests (not sure how easy it is)

Commit:
deps: multer@1.1.0

	modified:   package.json

deps: multer@1.1.0

jtwebman commented:
I don't see any file changes besides including the dependency?

ckousik commented:
I actually messed up when squashing commits....and now I don't know how to fix it. There's supposed to be quite a few changes

Commit:
deps: multer@1.1.0

	deleted:    core/server/middleware/ghost-busboy.js
	new file:   core/server/middleware/ghost-multer.js
	modified:   core/server/middleware/index.js
	modified:   core/server/routes/api.js
	modified:   package.json

ErisDS commented:
Time to get familiar with the reflog!

`git reflog -20` will show you the last 20 states that git knows about:

![](http://puu.sh/mMFvw.png)

Here you can see that each commit gets a `HEAD@{#}` reference. So if I wanted to go back before my rebase, I would need `HEAD@{3}`... you can see the next command would be `git reset HEAD@{3}`.

`git reset HEAD@{#}` will reset you back to the corresponding state.

This should bring your changes back

ErisDS commented:
FYI - this PR seems to have 2 commits still.

I have some other feedback but it's more relevant to the discussion I think, so I'll write it on the issue.

ErisDS commented:
https://github.com/TryGhost/Ghost/issues/6362#issuecomment-179309604

If we do press ahead with this change, the i18n keys for busboy should also be removed, or they should be changed to generic file upload keys and the equivalent error handling should be done.

ErisDS commented:
I left some comments on the issue and there has been further discussion but not yet any further discussion around the implementation in this PR. Going to close this for now - it is linked to the issue so we can use it for discussion still, but it needs some more discussion & updates before it could be merged.
